FSW/src/util/hammingBlock.cpp

```diff
@@ -0,0 +1,269 @@
+/* hammingBlock.cpp defines the HammingBlock class
```

Comment:
im gonna have to mostly trust you on this logic, we can talk through it in the meeting if there are any particular sections you're worried about though I would be happy to review them in detail